crates/primitives/src/env.rs

```diff
@@ -45,6 +45,14 @@ pub struct TxEnv {
     pub chain_id: Option<u64>,
     pub nonce: Option<u64>,
     pub access_list: Vec<(B160, Vec<U256>)>,
+    #[cfg(feature = "optimism")]
+    pub source_hash: Option<B256>,
+    #[cfg(feature = "optimism")]
+    pub mint: Option<u128>,
+    #[cfg(feature = "optimism")]
+    pub is_system_transaction: Option<bool>,
+    #[cfg(feature = "optimism")]
+    pub l1_cost: Option<U256>,
```

Comment:
Calling this out - having `l1_cost` in the env is a bit of a workaround. In order to compute the L1 cost of a transaction on OP, we need to have access to the full 2718 encoded tx within `revm`, which felt leaky. Instead of computing the L1 cost within revm, I've offloaded this responsibility to the processes that fill the `TxEnv`.

It may make more sense to just pass the enveloped encoding here rather than the L1 cost - we need to load the L1 block information within `revm` anyways in order to determine the fee that heads to the base fee vault, so doing this could save the parent process an extra database load of the `L1Block` account. Open to opinions here 😄

Reply:
We decided to switch to accept the enveloped tx bytes here instead (implemented in https://github.com/anton-rs/op-revm/pull/16).

The reason to go this route is that specifying the `l1_cost` here means downstream applications/users will need to compute the l1 tx cost themselves using the l1 block info. Since this can, and is done internally, exposing this computation is leaky.

Instead, we can accept the 2718 enveloped tx bytes and then compute the l1 cost internally. This a) reduces the amount of work that users have to perform and b) reduces "dead code" in the form of the L1 block info l1 cost computation already available inside the `optimism` module in the `revm` crate.

Ultimately, either method works but this should be easier to with 😄